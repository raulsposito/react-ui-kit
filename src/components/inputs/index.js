import TextField from './textField'
import Password from './password'
import NumberField from './numberField'
import TextArea from './textArea'

export {
  TextField,
  Password,
  NumberField,
  TextArea
}
