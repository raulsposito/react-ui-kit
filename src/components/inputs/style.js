import styled from 'styled-components'

export const StyledInputWrapper = styled.div`
  width: 100%;

  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-start;

  position: relative;
`

export const StyledInputWrapperHelper = styled.div`
  width: 100%;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
  flex-wrap: wrap;

  position: relative;
`
