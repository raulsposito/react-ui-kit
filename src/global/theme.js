// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
