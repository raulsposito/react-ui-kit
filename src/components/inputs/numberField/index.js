import React, { useState } from 'react'
import PropTypes from 'prop-types'

import MyPropTyes from '@global/propTypes'

import { INPUT_TYPES } from '../utils/constants'
import { validateField } from '../utils/validation'
import { StyledInputWrapper } from '../style'

import { handleTextColorState } from '../utils/handlers'

import InputLabel from '../components/inputLabel'
import InputContainer from '../components/inputContainer'
import InputHelperText from '../components/inputHelperText'
import InputAdornment from '../components/inputAdornment'

const NumberField = ({
  name,
  color,
  label,
  helperText,
  onChange,
  placeholder,
  value,
  disabled,
  error,
  required
}) => {
  const [focused, setFocused] = useState(false)
  const [fieldError, setFieldError] = useState(error)

  const handleError = () => {
    const type = INPUT_TYPES.NUMBER
    const inputError = validateField({ type, value, required })

    if (!inputError.isValid) {
      setFieldError(inputError.message)
    } else {
      setFieldError(error)
    }
  }

  const handleFocus = () => {
    setFocused(true)
  }

  const handleBlur = () => {
    setFocused(false)
    handleError()
  }

  const handleChange = inputValue => onChange(name, Number(inputValue))
  const incrementValue = () => handleChange(Number(value) + 1)
  const decreaseValue = () => handleChange(Number(value) - 1)

  return (
    <StyledInputWrapper>
      <InputLabel
        text={label}
        color={handleTextColorState(fieldError, focused, color)}
        name={name}
        disabled={disabled}
        error={fieldError}
        required={required}
      />
      <InputContainer
        type='number'
        value={value}
        placeholder={placeholder}
        onChange={e => handleChange(e.target.value)}
        onFocus={handleFocus}
        onBlur={handleBlur}
        name={name}
        disabled={disabled}
        error={fieldError}
        focused={focused}
        required={required}
        color={color}
        startAdornment={(
          <InputAdornment
            onClick={!disabled && decreaseValue}
            position='start'
            disabled={disabled}
            icon='expand_more'
            color={fieldError ? 'error' : 'system'}
          />
            )}
        endAdornment={(
          <InputAdornment
            onClick={!disabled && incrementValue}
            position='end'
            disabled={disabled}
            icon='expand_less'
            color={fieldError ? 'error' : 'system'}
          />
          )}
      />
      <InputHelperText
        text={fieldError || helperText}
        color={handleTextColorState(fieldError, focused, color)}
        disabled={disabled}
      />
    </StyledInputWrapper>
  )
}

NumberField.propTypes = {
  name: PropTypes.string,
  color: MyPropTyes.color,
  label: PropTypes.string,
  helperText: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  value: PropTypes.number,
  disabled: PropTypes.bool,
  error: PropTypes.string,
  required: PropTypes.bool
}

export default NumberField
