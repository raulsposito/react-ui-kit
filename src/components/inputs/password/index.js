import React, { useState } from 'react'
import PropTypes from 'prop-types'

import MyPropTyes from '@global/propTypes'

import { INPUT_TYPES } from '../utils/constants'
import { validateField } from '../utils/validation'
import { StyledInputWrapper } from '../style'

import { handleTextColorState } from '../utils/handlers'

import InputLabel from '../components/inputLabel'
import InputContainer from '../components/inputContainer'
import InputHelperText from '../components/inputHelperText'
import InputAdornment from '../components/inputAdornment'

const Password = ({ name, color, label, helperText, onChange, placeholder, value, disabled, error, required }) => {
  const [focused, setFocused] = useState(false)
  const [passwordVisibility, setPasswordVisibility] = useState(false)
  const [fieldError, setFieldError] = useState(error)

  const handleError = () => {
    const type = INPUT_TYPES.PASSWORD
    const inputError = validateField({ type, value, required })

    if (!inputError.isValid) {
      setFieldError(inputError.message)
    } else {
      setFieldError(error)
    }
  }

  const handleFocus = () => {
    setFocused(true)
  }

  const handleBlur = () => {
    setFocused(false)
    handleError()
  }

  const handleChange = e => onChange(name, e.target.value)

  const handlePasswordVisibility = () => {
    setPasswordVisibility(!passwordVisibility)
  }

  return (
    <StyledInputWrapper>
      <InputLabel
        text={label}
        color={handleTextColorState(fieldError, focused, color)}
        name={name}
        disabled={disabled}
        error={fieldError}
        focused={focused}
        required={required}
      />
      <InputContainer
        type={passwordVisibility ? 'text' : 'password'}
        value={value}
        placeholder={placeholder}
        onChange={handleChange}
        onFocus={handleFocus}
        onBlur={handleBlur}
        name={name}
        disabled={disabled}
        error={fieldError}
        focused={focused}
        required={required}
        color={color}
        endAdornment={(
          <InputAdornment
            onClick={!disabled ? handlePasswordVisibility : null}
            position='end'
            disabled={disabled}
            icon={!passwordVisibility ? 'visibility' : 'visibility_off'}
            color={fieldError ? 'error' : 'system'}
          />
        )}
      />
      <InputHelperText
        text={fieldError || helperText}
        color={handleTextColorState(fieldError, focused, color)}
        disabled={disabled}
        error={fieldError}
        focused={focused}
      />
    </StyledInputWrapper>
  )
}

Password.propTypes = {
  name: PropTypes.string,
  color: MyPropTyes.color,
  label: PropTypes.string,
  helperText: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.any,
  disabled: PropTypes.bool,
  error: PropTypes.string,
  required: PropTypes.bool
}

export default Password
