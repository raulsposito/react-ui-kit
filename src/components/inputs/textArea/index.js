import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'

import MyPropTyes from '@global/propTypes'

import { StyledInputWrapper, StyledInputWrapperHelper } from '../style'

import { handleTextColorState } from '../utils/handlers'

import InputLabel from '../components/inputLabel'
import InputHelperText from '../components/inputHelperText'
import InputCharacterCounter from '../components/inputCharacterCounter'
import InputAreaContainer from '../components/inputAreaContainer'
import { validateTextArea } from '../utils/validation'

const TextArea = props => {
  const {
    name,
    rows,
    color,
    label,
    value,
    error,
    resize,
    onChange,
    disabled,
    required,
    maxLength,
    helperText,
    placeholder
  } = props

  const [focused, setFocused] = useState(false)
  const [fieldError, setFieldError] = useState(error)
  const [characterCount, setCharacterCount] = useState(value.length)

  useEffect(() => {
    setCharacterCount(value.length)
  }, [value])

  const handleError = () => {
    const inputError = validateTextArea({ value, maxLength, required })

    if (!inputError.isValid) {
      setFieldError(inputError.message)
    } else {
      setFieldError(error)
    }
  }

  const handleFocus = () => {
    setFocused(true)
  }

  const handleBlur = () => {
    setFocused(false)
    handleError()
  }

  const handleChange = e => onChange(name, e.target.value)

  return (
    <StyledInputWrapper>
      <InputLabel
        text={label}
        color={handleTextColorState(fieldError, focused, color)}
        name={name}
        disabled={disabled}
        error={fieldError}
        required={required}
      />

      <InputAreaContainer
        value={value}
        name={name}
        placeholder={placeholder}
        required={required}
        onChange={handleChange}
        onFocus={handleFocus}
        onBlur={handleBlur}
        disabled={disabled}
        error={fieldError}
        maxLength={maxLength}
        rows={rows}
        resize={resize}
      />
      <StyledInputWrapperHelper>
        <InputCharacterCounter
          currentCharacterCount={characterCount}
          maxLength={maxLength}
          disabled={disabled}
        />
        <InputHelperText
          text={fieldError || helperText}
          color={handleTextColorState(fieldError, focused, color)}
          disabled={disabled}
        />
      </StyledInputWrapperHelper>
    </StyledInputWrapper>
  )
}

TextArea.propTypes = {
  name: PropTypes.string,
  color: MyPropTyes.color,
  label: PropTypes.string,
  helperText: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.any,
  disabled: PropTypes.bool,
  error: PropTypes.string,
  required: PropTypes.bool,
  maxLength: PropTypes.number,
  rows: PropTypes.number,
  resize: PropTypes.bool
}

export default TextArea
