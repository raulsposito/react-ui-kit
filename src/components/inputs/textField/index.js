import React, { useState } from 'react'
import PropTypes from 'prop-types'

import MyPropTyes from '@global/propTypes'

import { BASIC_INPUT_TYPES } from '@components/inputs/utils/constants'
import { validateField } from '../utils/validation'
import { StyledInputWrapper } from '../style'

import { handleTextColorState } from '../utils/handlers'

import InputLabel from '../components/inputLabel'
import InputContainer from '../components/inputContainer'
import InputHelperText from '../components/inputHelperText'
import InputAdornment from '../components/inputAdornment'

const TextField = ({ name, color, label, helperText, onChange, placeholder, type, value, endAdornment, startAdornment,
  disabled, error, required }) => {
  const [focused, setFocused] = useState(false)
  const [fieldError, setFieldError] = useState(error)
  const errorAdornment = <InputAdornment position='end' icon='error' color='error' />

  const handleError = () => {
    const inputError = validateField({ type, value, required })

    if (!inputError.isValid) {
      setFieldError(inputError.message)
    } else {
      setFieldError(error)
    }
  }

  const handleFocus = () => {
    setFocused(true)
  }

  const handleBlur = () => {
    setFocused(false)
    handleError()
  }

  const handleChange = e => onChange(name, e.target.value)

  return (
    <StyledInputWrapper>
      <InputLabel
        text={label}
        color={handleTextColorState(fieldError, focused, color)}
        name={name}
        disabled={disabled}
        error={fieldError}
        focused={focused}
        required={required}
      />
      <InputContainer
        type={type}
        value={value}
        placeholder={placeholder}
        onChange={handleChange}
        onFocus={handleFocus}
        onBlur={handleBlur}
        name={name}
        startAdornment={startAdornment}
        endAdornment={fieldError ? errorAdornment : endAdornment}
        disabled={disabled}
        error={fieldError}
        focused={focused}
        required={required}
        color={color}
      />
      <InputHelperText
        text={fieldError || helperText}
        color={handleTextColorState(fieldError, focused, color)}
        disabled={disabled}
        error={fieldError}
        focused={focused}
      />
    </StyledInputWrapper>
  )
}

TextField.propTypes = {
  name: PropTypes.string,
  color: MyPropTyes.color,
  label: PropTypes.string,
  helperText: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  type: PropTypes.oneOf(Object.values(BASIC_INPUT_TYPES)),
  value: PropTypes.any,
  endAdornment: PropTypes.node,
  startAdornment: PropTypes.node,
  disabled: PropTypes.bool,
  error: PropTypes.string,
  required: PropTypes.bool
}

export default TextField
