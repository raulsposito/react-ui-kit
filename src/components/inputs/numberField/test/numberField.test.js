import React from 'react'
import { fireEvent, render, screen, waitFor } from '@testing-library/react'

import { colors } from '@global/theme'

import { NumberField } from '@components/inputs'

let value = 234
const handleChage = e => { value = e }

const mockData = {
  name: 'numberMock',
  label: 'Number Label',
  requiredLabel: 'Number Label *',
  placeholder: 'number placeholder',
  value: 'goblinShaman',
  error: 'Error Text',
  helperText: 'Helper Text',
  color: 'secondary',
  wrongNumber: 'gob',
  goodNumber: 1234,
  onChange: jest.fn()
}

describe('<NumberField />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<NumberField
      name={mockData.name}
      label={mockData.label}
      placeholder={mockData.placeholder}
      helperText={mockData.helperText}
    />)
    expect(asFragment()).toMatchSnapshot()
  })

  describe('Disabled', () => {
    it('is disabled', () => {
      render(
        <NumberField
          name={mockData.name}
          label={mockData.label}
          placeholder={mockData.placeholder}
          helperText={mockData.helperText}
          disabled
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      expect(input).toBeDisabled()
    })
  })

  describe('Required', () => {
    it('is required', () => {
      render(
        <NumberField
          name={mockData.name}
          label={mockData.label}
          placeholder={mockData.placeholder}
          helperText={mockData.helperText}
          required
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      const label = screen.queryByText(mockData.requiredLabel)

      expect(input).toBeRequired()
      expect(label).toBeInTheDocument()
    })
  })

  describe('Placeholder', () => {
    it('input has placeholder', () => {
      render(
        <NumberField
          name={mockData.name}
          label={mockData.label}
          placeholder={mockData.placeholder}
          helperText={mockData.helperText}
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      expect(input).toBeInTheDocument()
    })
  })

  describe('Name', () => {
    it('input has name', () => {
      render(
        <NumberField
          name={mockData.name}
          label={mockData.label}
          placeholder={mockData.placeholder}
          helperText={mockData.helperText}
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder, { name: mockData.name })
      const label = screen.queryByText(mockData.label).closest('label')

      expect(input).toBeInTheDocument()
      expect(label).toHaveAttribute('for', mockData.name)
    })
  })

  describe('Input event', () => {
    it('calls function on change', () => {
      render(<NumberField
        name={mockData.name}
        label={mockData.label}
        placeholder={mockData.placeholder}
        helperText={mockData.helperText}
        onChange={mockData.onChange}
      />)

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      fireEvent.change(input, { target: { value: 222 } })
      expect(mockData.onChange).toHaveBeenCalledTimes(1)
    })
  })

  describe('Input Type', () => {
    it('is number type', () => {
      render(
        <NumberField
          name={mockData.name}
          label={mockData.label}
          placeholder={mockData.placeholder}
          helperText={mockData.helperText}
        />
      )

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      expect(input).toHaveAttribute('type', 'number')
    })
  })

  describe('Handle State Color', () => {
    describe('Handle one state at time', () => {
      it('has not state changes', () => {
        render(
          <NumberField
            name={mockData.name}
            label={mockData.label}
            placeholder={mockData.placeholder}
            helperText={mockData.helperText}
          />
        )

        const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
        const label = screen.queryByText(mockData.label)
        const helper = screen.queryByText(mockData.helperText)

        expect(inputContainer).toHaveStyle({ borderColor: `${colors.system.regular}` })
        expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
        expect(label).toHaveStyle({ color: `${colors.contrast.regular}` })
        expect(helper).toHaveStyle({ color: `${colors.contrast.regular}` })
      })

      it('is disabled', () => {
        render(<NumberField
          name={mockData.name}
          label={mockData.label}
          placeholder={mockData.placeholder}
          helperText={mockData.helperText}
          disabled
        />)

        const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
        expect(inputContainer).toHaveStyle({ borderColor: `${colors.system.disabled}` })
        expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.disabled}` })
      })

      it('has error', () => {
        render(<NumberField
          name={mockData.name}
          label={mockData.label}
          placeholder={mockData.placeholder}
          helperText={mockData.helperText}
          error={mockData.error}
        />)

        const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
        const label = screen.queryByText(mockData.label)
        const helper = screen.queryByText(mockData.error)
        const leftArrow = screen.queryByText('expand_more')
        const rightArrow = screen.queryByText('expand_less')

        expect(inputContainer).toHaveStyle({ borderColor: `${colors.error.regular}` })
        expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
        expect(label).toHaveStyle({ color: `${colors.error.regular}` })
        expect(helper).toBeInTheDocument()
        expect(helper).toHaveStyle({ color: `${colors.error.regular}` })
        expect(leftArrow).toHaveStyle({ color: `${colors.error.regular}` })
        expect(rightArrow).toHaveStyle({ color: `${colors.error.regular}` })
      })

      it('has custom color', () => {
        render(<NumberField
          name={mockData.name}
          label={mockData.label}
          placeholder={mockData.placeholder}
          helperText={mockData.helperText}
          color={mockData.color}
        />)

        const input = screen.queryByPlaceholderText(mockData.placeholder)
        const label = screen.queryByText(mockData.label)
        const helper = screen.queryByText(mockData.helperText)

        fireEvent.focus(input)

        expect(helper).toHaveStyle({ color: `${colors[mockData.color].regular}` })
        expect(label).toHaveStyle({ color: `${colors[mockData.color].regular}` })
      })
    })

    describe('Handle multiple states at time', () => {
      it('is disabled and has custom color', () => {
        render(<NumberField
          name={mockData.name}
          label={mockData.label}
          placeholder={mockData.placeholder}
          helperText={mockData.helperText}
          disabled
          color={mockData.color}
        />)

        const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')

        expect(inputContainer).toHaveStyle({ borderColor: `${colors.system.disabled}` })
        expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.disabled}` })
      })

      it('has error and custom color', () => {
        render(<NumberField
          name={mockData.name}
          label={mockData.label}
          placeholder={mockData.placeholder}
          helperText={mockData.helperText}
          error={mockData.error}
          color={mockData.color}
        />)

        const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
        const label = screen.queryByText(mockData.label)
        const helper = screen.queryByText(mockData.error)
        const leftArrow = screen.queryByText('expand_more')
        const rightArrow = screen.queryByText('expand_less')

        expect(inputContainer).toHaveStyle({ borderColor: `${colors.error.regular}` })
        expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
        expect(label).toHaveStyle({ color: `${colors.error.regular}` })
        expect(helper).toHaveStyle({ color: `${colors.error.regular}` })
        expect(leftArrow).toHaveStyle({ color: `${colors.error.regular}` })
        expect(rightArrow).toHaveStyle({ color: `${colors.error.regular}` })
      })
    })
  })

  describe('Validations', () => {
    it('validation wrong', () => {
      render(<NumberField
        name={mockData.name}
        label={mockData.label}
        placeholder={mockData.placeholder}
        helperText={mockData.helperText}
        value={mockData.wrongNumber}
      />)

    // is not possible to enter a string value
    })

    it('priorize number validation error from error property', () => {
      render(<NumberField
        name={mockData.name}
        label={mockData.label}
        placeholder={mockData.placeholder}
        helperText={mockData.helperText}
        error={mockData.error}
        value={mockData.wrongNumber}
      />)

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
      const label = screen.queryByText(mockData.label)
      const helper = screen.queryByText(mockData.error)
      const leftArrow = screen.queryByText('expand_more')
      const rightArrow = screen.queryByText('expand_less')

      fireEvent.blur(input)
      expect(inputContainer).toHaveStyle({ borderColor: `${colors.error.regular}` })
      expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
      expect(label).toHaveStyle({ color: `${colors.error.regular}` })
      expect(helper).not.toBe(mockData.helperText)
      expect(helper).not.toBe(mockData.error)
      expect(helper).toHaveStyle({ color: `${colors.error.regular}` })
      expect(helper).toBeInTheDocument()
      expect(leftArrow).toHaveStyle({ color: `${colors.error.regular}` })
      expect(rightArrow).toHaveStyle({ color: `${colors.error.regular}` })
    })
  })

  describe('Controlled Component Behavior', () => {
    it('controlled value set', () => {
      render(<NumberField
        name={mockData.name}
        label={mockData.label}
        placeholder={mockData.placeholder}
        helperText={mockData.helperText}
        error={mockData.error}
        value={value}
        onChange={handleChage}
      />)

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      expect(Number(input.value)).toBe(value)
    })

    it('controlled value change', () => {
      render(<NumberField
        name={mockData.name}
        label={mockData.label}
        placeholder={mockData.placeholder}
        helperText={mockData.helperText}
        error={mockData.error}
        value={value}
        onChange={handleChage}
      />)

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      fireEvent.change(input, { target: { value: 23 } })
      waitFor(() => expect(Number(input.value)).toBe(value))
    })
  })
})
