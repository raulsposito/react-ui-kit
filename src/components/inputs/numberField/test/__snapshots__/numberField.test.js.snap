// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberField /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBQB TpSNq"
  >
    <label
      class="sc-pGacB liSCmd"
      for="numberMock"
    >
      <p
        class="sc-dlfnuX cALFVY"
        color="contrast"
      >
        Number Label 
      </p>
    </label>
    <div
      class="sc-kEjbQP NJzOK"
    >
      <div
        class="sc-dQoVA cosYkr"
      >
        <span
          class="sc-crrszt fQFtWD material-icons"
          color="system"
        >
          expand_more
        </span>
      </div>
      <input
        class="sc-jrAFXE hLRQfv"
        name="numberMock"
        placeholder="number placeholder"
        type="number"
        value=""
      />
      <div
        class="sc-dQoVA itcOIr"
      >
        <span
          class="sc-crrszt fQFtWD material-icons"
          color="system"
        >
          expand_less
        </span>
      </div>
    </div>
    <span
      class="sc-iqHYmW fBDEii"
    >
      <p
        class="sc-dlfnuX cALFVY"
        color="contrast"
      >
        Helper Text
      </p>
    </span>
  </div>
</DocumentFragment>
`;
